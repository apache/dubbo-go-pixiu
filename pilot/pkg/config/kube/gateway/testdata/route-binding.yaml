apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
  - name: foobar
    hostname: "*.foobar.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
  - name: same-namespace
    hostname: "*.same-namespace.example"
    port: 80
    protocol: HTTP
  - name: scope-route
    hostname: "*.scope-route.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All
      kinds:
      - kind: TCPRoute
  - name: namespace-selector
    hostname: "*.namespace-selector.example"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            istio.io/test-name-part: group
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: section-name-cross-namespace
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: foobar
  hostnames: ["alpha.foobar.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: same-namespace-valid
  namespace: istio-system
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: foobar
  - name: gateway
    namespace: istio-system
    sectionName: same-namespace
  rules:
  - backendRefs:
    - name: httpbin
      port: 81
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: same-namespace-invalid
  namespace: default
spec:
  parentRefs:
  - kind: Gateway
    name: gateway
    namespace: istio-system
    sectionName: same-namespace
  hostnames: ["foo.same.example"]
  rules:
  - backendRefs:
    - name: echo
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TCPRoute
metadata:
  # Should not generate anything, the protocol is HTTP
  name: wrong-protocol
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: foobar
  rules:
  - backendRefs:
    - name: httpbin
      port: 82
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: host-mismatch
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: foobar
  hostnames: ["no.match.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 84
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: bind-all
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  rules:
  - backendRefs:
    - name: httpbin
      port: 85
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: bind-cross-namespace
  namespace: group-namespace1
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: namespace-selector
  rules:
  - backendRefs:
    - name: httpbin
      port: 86
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: bind-cross-namespace
  namespace: group-namespace2
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: namespace-selector
  rules:
  - backendRefs:
    - name: httpbin
      port: 87
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: invalid-bind-cross-namespace
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
    sectionName: namespace-selector
  rules:
  - backendRefs:
    - name: httpbin
      port: 87

