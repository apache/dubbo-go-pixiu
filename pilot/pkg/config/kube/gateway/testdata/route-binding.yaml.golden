apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/foobar.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-foobar
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.foobar.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/same-namespace.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-same-namespace
  namespace: istio-system
spec:
  servers:
  - hosts:
    - istio-system/*.same-namespace.example
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/scope-route.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-scope-route
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.scope-route.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/namespace-selector.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-namespace-selector
  namespace: istio-system
spec:
  servers:
  - hosts:
    - group-namespace1/*.namespace-selector.example
    - group-namespace2/*.namespace-selector.example
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/section-name-cross-namespace.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: section-name-cross-namespace-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-foobar
  hosts:
  - alpha.foobar.example
  http:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/same-namespace-valid.istio-system
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: same-namespace-valid-0-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-same-namespace
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin.istio-system.svc.domain.suffix
        port:
          number: 81
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/bind-all.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: bind-all-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 85
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/bind-all.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: bind-all-1-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-foobar
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 85
  - route:
    - destination:
        host: httpbin.istio-system.svc.domain.suffix
        port:
          number: 81
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/bind-all.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: bind-all-2-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-scope-route
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 85
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/bind-cross-namespace.group-namespace1
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: bind-cross-namespace-0-istio-autogenerated-k8s-gateway
  namespace: group-namespace1
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-namespace-selector
  hosts:
  - '*'
  http:
  - route:
    - destination:
        host: httpbin.group-namespace1.svc.domain.suffix
        port:
          number: 86
  - route:
    - destination:
        host: httpbin.group-namespace2.svc.domain.suffix
        port:
          number: 87
---
